overlord/snapstate/snapstate.go

```diff
@@ -722,6 +722,9 @@ func doUpdate(st *state.State, names []string, updates []*snap.Info, params func
 			}
 			return nil, nil, err
 		}
+		if err := canSwitch(st, update.SnapName(), channel); err != nil {
```

Comment:
I think it would be fine to move this check to Update, and not here. For UpdateMany  channel is always snapst.Channel